pkg/cli/install/steps.go

```diff
@@ -0,0 +1,191 @@
+				if err != nil {
+					return false, fmt.Errorf("catalog source not found")
+				}
+				return pointer.Get(cs.Status.GRPCConnectionState).LastObservedState == "READY", nil
+			})
+		},
+	}
+}
+
+func (o *Install) waitForDeployment(ctx context.Context, name, namespace string) error {
+	o.l.Infof("Waiting for Deployment '%s' in namespace '%s'", name, namespace)
+	if err := o.kubeClient.WaitForRollout(ctx, name, namespace); err != nil {
+		return err
+	}
+	o.l.Infof("Deployment '%s' in namespace '%s' is ready", name, namespace)
+	return nil
+}
+
+func (o *Install) installEverestHelmChart(ctx context.Context) error {
+	o.l.Info("Installing Everest Helm chart")
+	if err := o.helmInstaller.Install(ctx); err != nil {
+		return fmt.Errorf("could not install Helm chart: %w", err)
+	}
+	return nil
+}
+
+// TODO: remove this after we move to the victoria-metrics Helm chart.
+func (o *Install) installVMOperator(ctx context.Context) error {
+	// Approve Victoriametrics operator install plan.
+	return backoff.Retry(func() error {
+		channel := "stable-v0"
+		name := "victoriametrics-operator"
+		return o.kubeClient.InstallOperator(ctx, kubernetes.InstallOperatorRequest{
+			Channel:                channel,
+			Name:                   name,
+			Namespace:              common.MonitoringNamespace,
+			CatalogSource:          common.PerconaEverestCatalogName,
+			CatalogSourceNamespace: kubernetes.OLMNamespace,
+			OperatorGroup:          common.MonitoringNamespace,
+			InstallPlanApproval:    olmv1alpha1.ApprovalManual,
+		})
+	}, backoff.NewConstantBackOff(backoffInterval),
+	)
+}
+
+func (o *Install) provisionDBNamespace(ver string, namespace string) steps.Step {
+	return steps.Step{
+		Desc: fmt.Sprintf("Provisioning DB namespace '%s'", namespace),
+		F: func(ctx context.Context) error {
+			if err := o.createNamespace(ctx, namespace); err != nil {
+				return err
+			}
+			chartDir := ""
+			if o.config.ChartDir != "" {
+				chartDir = path.Join(o.config.ChartDir, dbNamespaceSubChartPath)
+			}
+
+			values := helmutils.MustMergeValues(
+				o.getDBNamespaceInstallValues(),
+				helm.ClusterValues(o.clusterType),
+			)
+
+			installer := helm.Installer{
+				ReleaseName:            namespace,
+				ReleaseNamespace:       namespace,
+				Values:                 values,
+				CreateReleaseNamespace: false,
+			}
+			if err := installer.Init(o.config.KubeconfigPath, helm.ChartOptions{
+				Directory: chartDir,
+				URL:       o.config.RepoURL,
+				Name:      helm.EverestDBNamespaceChartName,
+				Version:   ver,
+			}); err != nil {
+				return fmt.Errorf("could not initialize Helm installer: %w", err)
+			}
+
+			o.l.Infof("Installing DB namespace Helm chart in namespace ", namespace)
+			if err := installer.Install(ctx); err != nil {
+				return fmt.Errorf("could not install Helm chart: %w", err)
+			}
+
+			if err := o.installDBOperators(ctx, namespace); err != nil {
+				return err
+			}
```

Comment:
I'm quite confused about this :see_no_evil: 
I thought that the DB namespace sub chart would handle all logic of installing the DB operators. Why are we managing the Subscriptions/InstallPlans manually here?

Reply:
Yeah, the sub-chart can actually handle this.. initially I didn't want to run any hooks at all in the CLI so that we could show the steps in detail to the user. For example, with the monitoring, I wanted it to be a separate step rather than being a part of the "Installing Helm chart". (now its a different story since with [this](https://github.com/percona/percona-helm-charts/pull/416) PR all hooks are gone from the core chart)

But now that I think about it, maybe we can allow the use of hooks in the sub-chart since the DB provisioning happens in 1 step.

Another reason is that if I enable hooks in the sub-chart, it will also run the clean-up hooks (remove DB, BackupStorage, etc.) and we won't be able to show those steps in the CLI. But later I figured a workaround to selectively disable hooks..

So I think we can get rid of this code afterall and let the sub-chart hooks handle it. Thanks for noticing this :)

Reply:
Updated